typing/shape_reduce.ml

```diff
@@ -323,7 +331,7 @@ end) = struct
       Unresolved (read_back env nf)
     else match nf with
       | { desc = NAlias _; approximated = false; _ } ->
-          get_aliases_uids (read_back env nf)
+          read_back_aliases_uids env nf
```

Comment:
I wonder if the code would look nicer if we had

```ocaml
type result =
   ...
  | Resolved_alias of uid.t * result
```

so that we would track alias chains also in the "approximated" and other cases.